source/ui/Components/Settings/default-settings.json

```diff
@@ -1,4 +1,6 @@
-{
+{
     "snapshotLocation": null,
-    "theme": "default"
+    "theme": "default",
+    "removeSnapshotsPermanently": null,

```

Comment:
I have fixed this by requiring at least one visible change to a setting before it is marked as having changed (such that even if the initial value is different from the setting, because in this case the setting must be a boolean but is really `undefined`/`null`, this is not marked as a change)